# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/frontend/ParseNode.cpp
# Commit: 73cc5fa0e0bd
# Full Hash: 73cc5fa0e0bdd7a1d6b0f8450b1541c4bc53c333
# Author: Tooru Fujisawa <arai_a@mac.com>
# Date: 2020-11-14 09:46:25
# Regressor Bug: 1675241
# File Overlap Count: 1
# Description:
#   Bug 1675241 - Part 9: Use TaggedParserAtomIndex in RegExpStencil.atom_. r=tcampbell
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D96916
# ==============================================================================

diff -r f505b8fb8f8b -r 73cc5fa0e0bd js/src/frontend/ParseNode.cpp
--- a/js/src/frontend/ParseNode.cpp	Sat Nov 14 00:03:07 2020 +0000
+++ b/js/src/frontend/ParseNode.cpp	Sat Nov 14 00:03:20 2020 +0000
@@ -399,13 +399,16 @@
 
 RegExpObject* RegExpStencil::createRegExp(
     JSContext* cx, CompilationAtomCache& atomCache) const {
-  RootedAtom atom(cx, atom_->toExistingJSAtom(cx, atomCache));
+  RootedAtom atom(cx, atomCache.getExistingAtomAt(cx, atom_));
   return RegExpObject::createSyntaxChecked(cx, atom, flags_, TenuredObject);
 }
 
 RegExpObject* RegExpStencil::createRegExpAndEnsureAtom(
-    JSContext* cx, CompilationAtomCache& atomCache) const {
-  RootedAtom atom(cx, atom_->toJSAtom(cx, atomCache));
+    JSContext* cx, CompilationAtomCache& atomCache,
+    CompilationStencil& stencil) const {
+  const ParserAtom* parserAtom = stencil.getParserAtomAt(cx, atom_);
+  MOZ_ASSERT(parserAtom);
+  RootedAtom atom(cx, parserAtom->toJSAtom(cx, atomCache));
   if (!atom) {
     return nullptr;
   }
@@ -415,7 +418,8 @@
 RegExpObject* RegExpLiteral::create(JSContext* cx,
                                     CompilationAtomCache& atomCache,
                                     CompilationStencil& stencil) const {
-  return stencil.regExpData[index_].createRegExpAndEnsureAtom(cx, atomCache);
+  return stencil.regExpData[index_].createRegExpAndEnsureAtom(cx, atomCache,
+                                                              stencil);
 }
 
 bool js::frontend::IsAnonymousFunctionDefinition(ParseNode* pn) {